core/os/device/deviceinfo/cc/android/query.cpp

```diff
@@ -328,6 +328,9 @@ bool createContext() {
 
   GET_STRING_PROP("ro.build.version.release", gContext.mOSName);
   GET_INT_PROP("ro.build.version.sdk", gContext.mOSVersion);
+  int previewSdk = 0;
+  GET_INT_PROP("ro.build.version.preview_sdk", previewSdk);
+  gContext.mOSVersion += previewSdk;
```

Comment:
mOSVersion is used in the switch statement a little later. Check how mOSVersionMajor/Minor is used and see if it can break anything and check if we have any weird cases where we have different ways to get the OS version.  I'd imagine you will at least have to add a case 30 for Android 11.

Reply:
Yeah. Peiyong has put up a PR for that. I'll merge this in only after that merges in.